Ooui/EventTarget.cs

```diff
@@ -14,7 +14,9 @@ public abstract class EventTarget : INotifyPropertyChanged
         readonly Dictionary<string, List<TargetEventHandler>> eventListeners =
             new Dictionary<string, List<TargetEventHandler>> ();
 
-        public string Id { get; protected set; } = GenerateId ();
+        public long PersistentId { get; } = GenerateId ();
```

Comment:
For the replace code to work I needed an element Identity that can't be changed. Also I wanted something faster to compare than strings.

Reply:
I don't like the IDs floating around. IDs are immutable by semantics. Yes, C# is failing me here, but I do not want to introduce another ID.